subprojects/core/src/integTest/groovy/org/gradle/api/provider/ProviderUsageInKotlinIntegrationTest.groovy

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.provider
+
+import org.gradle.integtests.fixtures.KotlinScriptIntegrationTest
+import org.gradle.util.Requires
+import spock.lang.Ignore
+
+import static org.gradle.util.TestPrecondition.FIX_TO_WORK_ON_JAVA9
+import static org.gradle.util.TestPrecondition.NOT_JDK_IBM
+import static org.gradle.util.TextUtil.normaliseFileSeparators
+
+@Ignore("Fails on CI for unknown reason")
+@Requires([FIX_TO_WORK_ON_JAVA9, NOT_JDK_IBM])
```

Comment:
let's add something like `@Requires(KOTLIN_SCRIPT)` so that when we fix Kotlin support for Java 9 or IBM JDK we can change the definition in a single place.